packages/react-storage/src/components/StorageBrowser/views/hooks/useSearch.ts

```diff
@@ -0,0 +1,77 @@
+import React from 'react';
+import debounce from 'lodash/debounce';
+
+const DEFAULT_DELAY = 300;
+export interface InitialSearchValues<T> {
+  searchKey: keyof T;
+  searchTerm?: string;
+  debounceDelay?: number;
+}
+
+interface UseSearchProps<T> {
+  initialValues: InitialSearchValues<T>;
+  items: T[];
+  onSearch: (term: string, includeSubfolders?: boolean) => void;
+}
+
+interface UseSearch<T> {
+  filteredItems: T[];
+  searchTerm: string;
+  handleSearch: (term: string, includeSubfolders?: boolean) => void;
```

Comment:
I understand the intent here of distinguishing between the onSearch from the props and the wrapped version coming out of this hook but I wonder if it's actually necessary to sacrifice the `onX` convention. Imo it wouldn't actually be that bad to just vend the debounced `onSearch` as also `onSearch` from this hook. Ultimately, the consumer of the hook doesn't need to know there's even a difference.

Alternatively, if that is really too unpalatable, something more pointed like `debouncedOnSearch` would be ok?

Reply:
I'm not opposed to using `onSearch`, but you're right about the intention. I can change it to be consistent with other hooks